hdl/jpegenc_v2/myhdl/rgb2ycbcr.py

```diff
@@ -0,0 +1,78 @@
+#!/bin/python
+from myhdl import *
+from random import randrange
+
+Y1, Y2, Y3 = 4899, 9617, 1868
+CB1, CB2, CB3 = -2764, -5428, 8192
+CR1, CR2, CR3 = 8192, -6860, -1332
+Y_OFFSET, CB_OFFSET, CR_OFFSET = 0, 2097152, 2097152
+
+def rgb2ycbcr(y, cb, cr, r, g, b, clk, reset):
```

Comment:
Did the class look something like this?

```python
class RGB(object):
    def __init__(self, color_depth=(8,8,8)):
        cd = color_depth
        self.red = Signal(intbv(0)[cd[0]:])
        self.green = Signal(intbv(0)[cd[1]:]) 
        self.blue = Signal(intbv(0)[cd[2]:]) 
```
https://github.com/cfelton/minnesota/blob/master/mn/cores/video/vga/_vga_intf.py